### ISSUE TITLE
Implement EuclideanDistance for all combinations of geometry types

### ISSUE THREAD

**Issue by michaelkirk** [georust/geo#979]
Currently we can compute EucldieanDistance for lots of things, but not _everything_.

Ultimately, we want to add a 
```
impl EuclideanDistance for Geometry<T> {
    fn euclidean_distance(&self, other: &Geometry<T>) -> T
}
```

Implementing that should lead you down a path of compiler errors leading to any other missing combinations.

**Comment by JosiahParry** [comment]
This would be _very_ helpful! Right now I'm trying to write a wrapper for calculating a distance matrix between two vectors of Geometrys. But the branching of types  is more than my smooth brain can handle!

**Comment by urschrei** [comment]
Closed by #1029. Thanks @mcassels!
